File: AddAPearl.Domain/Item.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AddAPearl.Domain
{
    public class Item
    {
        public int ItemId { get; set; }

        public Product Product { get; set; }

        public int ProductId { get; set; }

        public string ItemName { get; set; }

        public decimal? Price { get; set; }

        public decimal PurchasePrice { get; set; }

        public DateTime? PurchaseDate { get; set; }

        public string Description { get; set; }

        public Customer Owner { get; set; }

        public int OwnerId { get; set; }

        public Customer Customer { get; set; }

        public int CustomerId { get; set; }

        public ICollection<SubItem> SubItems { get; set; }

        public decimal? Rating { get; set; }
    }
}

File: AddAPearl.Web/src/app/shared/models/item.ts
import {
	ICustomer,
	IProduct,
	ISubItem,
} from './index';

/* Defines the Item entity */
export interface IItem {
	itemId: number;
	productId: number;
	product: IProduct;
	itemName: string;
	price: number;
	purchasPrice: number;
	purchasDate: string;
	description: string;
	ownerId: number;
	owner: ICustomer;
	customerId: number;
	customer: ICustomer;
	subItems: ISubItem[];
	rating: number;
}

export class Item implements IItem {
	constructor(
		public itemId: number,
		public productId: number,
		public product: IProduct,
		public itemName: string,
		public price: number,
		public purchasPrice: number,
		public purchasDate: string,
		public description: string,
		public ownerId: number,
		public owner: ICustomer,
		public customerId: number,
		public customer: ICustomer,
		public subItems: ISubItem[],
		public rating: number,
	) {
		// Any other stuff to do during construction...
	}
}
